rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -260,18 +261,23 @@ fn build_batch_from_indices(
     // this is just for symmetry of the code below.
     let right = vec![right.clone()];
 
+    let (primary_is_left, primary, secondary) = match join_type {
```

Comment:
Maybe this could be called `outer_is_left`, `outer`, and `inner` to use standard terms rather than `primary` and `secondary`